File: iOS/iOS开发指南(从Hello World到App Store上架)/swift/11/11.4.2/MyNotes/NoteDAO.m
//
//  NoteDAO.m
//  MyNotes
//
//  Created by andyron on 2016/12/20.
//  Copyright © 2016年 andyron. All rights reserved.
//

#import "NoteDAO.h"

@implementation NoteDAO

static NoteDAO *sharedSingleton = nil;

+ (NoteDAO*)sharedManager {
    static dispatch_once_t once;
    dispatch_once(&once, ^{
        sharedSingleton = [[self alloc] init];
        
        NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
        [dateFormatter setDateFormat:@"yyyy-MM-dd HH:mm:ss"];
        
        NSDate *date1 = [dateFormatter dateFromString:@"2016-12-20 01:48:57"];
        Note *note1 = [[Note alloc] initWithDate:date1 content:@"Welcome to MyNote"];
        
        NSDate *date2 = [dateFormatter dateFromString:@"2016-12-21 01:50:19"];
        Note *note2 = [[Note alloc] initWithDate:date2 content:@"欢迎使用MyNote。"];
        
        sharedSingleton.listData = [[NSMutableArray alloc] init];
        [sharedSingleton.listData addObject:note1];
        [sharedSingleton.listData addObject:note2];

    });
    return sharedSingleton;
}

-(int)create:(Note *)model {
    [self.listData addObject:model];
    return 0;
}

-(int)remove:(Note *)model {
    for (Note *note in self.listData) {
        if ([note.date isEqualToDate:model.date]) {
            [self.listData removeObject:note];
        }
    }
    return 0;
}

-(int)modify:(Note *)model {
    for (Note *note in self.listData) {
        if ([note.date isEqualToDate:model.date]) {
            note.content = model.content;
            break;
        }
    }
    return 0;
}
- (NSMutableArray*) findAll {
    return self.listData;
}

-(Note *) findById:(Note *)model {
    for (Note *note in self.listData) {
        if ([note.date isEqualToDate:model.date]) {
            return note;
        }
    }
    return nil;
}



@end

File: iOS/iOS开发指南(从Hello World到App Store上架)/swift/11/11.4.1/MyNotes/NoteDAO.swift
//
//  NoteDAO.swift
//  MyNotes
//
//  Created by andyron on 2016/12/19.
//  Copyright © 2016年 andyron. All rights reserved.
//

import Foundation

class NoteDAO {
    var listData: NSMutableArray!
    
    static let sharedInstance: NoteDAO = {
        let instance = NoteDAO()
        
        let dateFormatter: DateFormatter = DateFormatter()
        dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
        let date1: Date = dateFormatter.date(from: "2016-12-19 22:56:14")!
        let note1: Note = Note(date: date1, content: "Welcome to MyNote")
        
        let date2: Date = dateFormatter.date(from: "2016-12-20 23:00:19")!
        let note2: Note = Note(date: date2, content: "欢迎使用MyNote。")
        
        instance.listData = NSMutableArray()
        instance.listData.addObjects(from: [note1, note2])
        return instance
    }()
    
    func create(model: Note) -> Int {
        self.listData.addObjects(from: [model])
        return 0
    }
    
    func remove(model: Note) -> Int {
        for note in self.listData {
            let note2 = note as! Note
            if note2.date == model.date {
                self.listData.remove(note)
            }
        }
        return 0
    }
    
    func modify(model: Note) -> Int {
        for note in self.listData {
            let note2 = note as! Note
            if note2.date == model.date {
                note2.content = model.content
                break
            }
        }
        return 0
    }
    
    func findAll() -> NSMutableArray {
        return self.listData
    }
    
    func findById(model: Note) -> Note? {
        for note in self.listData {
            let note2 = note as! Note
            if note2.date == model.date {
                return note2
            }
        }
        return nil
    }
    
}
